pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1520,25 +1533,32 @@ func TestMergePullSecrets(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			fakeClient := fake.NewFakeClient(test.existingCD...)
+			if test.globalPullSecret != "" {
+				globalPullSecret := createGlobalPullSecretObj(corev1.SecretTypeDockerConfigJson, globalPullSecret, corev1.DockerConfigJsonKey, test.globalPullSecret)
+				test.existingObjs = append(test.existingObjs, globalPullSecret)
+			}
+			fakeClient := fake.NewFakeClient(test.existingObjs...)
 			rcd := &ReconcileClusterDeployment{
 				Client:                        fakeClient,
 				scheme:                        scheme.Scheme,
 				logger:                        log.WithField("controller", "clusterDeployment"),
 				remoteClusterAPIClientBuilder: testRemoteClusterAPIClientBuilder,
 			}
 
-			localSecretObject := testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigJsonKey, test.localPullSecret)
-			err := rcd.Create(context.TODO(), localSecretObject)
-			if err != nil {
-				t.Errorf("unexpected error: %v", err)
-				return
+			if test.localPullSecret != "" {
+				localSecretObject := testSecret(corev1.SecretTypeDockercfg, pullSecretSecret, corev1.DockerConfigJsonKey, test.localPullSecret)
+				err := rcd.Create(context.TODO(), localSecretObject)
```

Comment:
It would be simpler to add the local pull secret to the existing objects passed to the fake client when it is built, like is being done for the global pull secret.

Reply:
Agree